tests/data_context/migrator/conftest.py

```diff
@@ -1,5 +1,187 @@
+        self._anonymous_usage_stats_is_none = anonymous_usage_stats_is_none
+        self._include_datasources = include_datasources
+
+    @property
+    def _data_context_variables(self) -> StubUsageStats:
+        return StubUsageStats()
+
+    @property
+    def anonymous_usage_statistics(self) -> AnonymizedUsageStatisticsConfig:
+        return self.variables.anonymous_usage_statistics
+
+    @property
+    def data_context_id(self) -> str:
+        return self.DATA_CONTEXT_ID
+
+    @property
+    def variables(self) -> DataContextVariables:
+
+        # anonymous_usage_statistics set based on constructor parameters.
+        anonymous_usage_statistics: Optional[AnonymizedUsageStatisticsConfig]
+        if self._anonymous_usage_stats_is_none:
+            anonymous_usage_statistics = None
+        else:
+            anonymous_usage_statistics = AnonymizedUsageStatisticsConfig(
+                enabled=self._anonymous_usage_stats_enabled
+            )
+
+        config = DataContextConfig(
+            anonymous_usage_statistics=anonymous_usage_statistics
+        )
+        return EphemeralDataContextVariables(config=config)
+
+    @property
+    def _datasource_store(self):
+        return StubDatasourceStore()
+
+    @property
+    def datasources(self) -> Dict[str, Union[LegacyDatasource, BaseDatasource]]:
+        # Datasource is a dummy since we just want the DatasourceConfig from the store, not an
+        # actual initialized datasource.
+        if self._include_datasources:
+            return {"my_datasource": DummyDatasource()}
+        else:
+            return {}
+
+    @property
+    def checkpoint_store(self) -> StubCheckpointStore:
+        return StubCheckpointStore()
+
+    @property
+    def validations_store(self) -> StubValidationsStore:
+        return StubValidationsStore()
+
+    def list_expectation_suite_names(self) -> List[str]:
+        return ["my_suite"]
+
+    def get_expectation_suite(self, name: str) -> ExpectationSuite:
+        return ExpectationSuite(expectation_suite_name=name)
+
+    def list_checkpoints(self) -> List[str]:
+        return ["my_checkpoint"]
+
+    def list_profilers(self) -> List[str]:
+        return ["my_profiler"]
+
+    def get_profiler(self, name: str) -> RuleBasedProfiler:
+        return RuleBasedProfiler(name, config_version=1.0, rules={})
+
+
+@pytest.fixture
+def stub_base_data_context_factory() -> Callable:
+    def _create_stub_base_data_context(
+        anonymous_usage_stats_enabled: bool = True,
+        anonymous_usage_stats_is_none: bool = False,
+        include_datasources: bool = True,
+    ) -> StubBaseDataContext:
+        return StubBaseDataContext(
+            anonymous_usage_stats_enabled=anonymous_usage_stats_enabled,
+            anonymous_usage_stats_is_none=anonymous_usage_stats_is_none,
+            include_datasources=include_datasources,
+        )
+
+    return _create_stub_base_data_context
```

Comment:
I'm finding this pattern to be quite effective when working with customizable stubs - I've modified the rest of the fixtures below to leverage this

Reply:
Ah crap - I like the pattern but this code all changed in https://github.com/great-expectations/great_expectations/pull/6107 to use dependency injection to create an empty config. Could you merge those changes in?

Reply:
Prefer your approach - updated across the uses